packages/svelte/src/internal/client/dom/blocks/html.js

```diff
@@ -29,7 +31,24 @@ export function html(anchor, get_value, svg, mathml) {
 
 		effect = branch(() => {
 			if (hydrating) {
-				assign_nodes(get_start(), hydrate_nodes[hydrate_nodes.length - 1]);
+				var next = hydrate_next();
+				var last = next;
+
+				while (
+					next !== null &&
+					(next.nodeType !== 8 || /** @type {Comment} */ (next).data !== HYDRATION_ANCHOR)
+				) {
+					last = next;
+					next = /** @type {TemplateNode} */ (next.nextSibling);
+				}
+
+				if (next === null) {
+					w.hydration_mismatch();
+					throw HYDRATION_ERROR;
+				}
+
+				assign_nodes(hydrate_node, last);
```

Comment:
I thought that was a deliberate choice we'd previously made?

Reply:
yeah it's a deliberate choice we made - I just wanted to bring it up here in case the new way makes things faster, which means we _could_ investigate doing that. And if not, a DEV time warning similar to how we do it for the img src would be nice.

Reply:
I think the wrinkle would be that if you had something like this...

```svelte
{@html '<div class=blah></div>'}
```

...the browser would add quotes around it, so if you concatenated all the nodes you found you'd get a mismatch...

```
<div class=blah></div>
<div class="blah"></div>
```

...unless you threw both things into a `<template>` and compared their shapes, which seems annoying.

One thing we _could_ do is (at dev time) hash the string and put it in the opening hydration comment, so that the server-rendered HTML looks like this:

```html
<!--khs73f--><div class="blah"></div><!---->
```

Then, at hydration time, you'd hash the string and compare. Will open an issue for that.

Reply:
We can use #12333 for that, where I also noted that we need docs around this

Reply:
https://github.com/sveltejs/svelte/issues/12362